flatland/envs/rail_grid_transition_map.py

```diff
@@ -166,23 +104,38 @@ def _check_action_new(self, action: RailEnvActions, position: IntVector2D, direc
             # new_direction will be the only valid transition
             # - take only available transition
             new_direction = fast_argmax(possible_transitions)
-            transition_valid = True
-            return new_direction, transition_valid
+            if action == RailEnvActions.MOVE_LEFT and new_direction != (direction - 1) % 4:
+                action = RailEnvActions.MOVE_FORWARD
```

Comment:
Return DO_NOTHING instead of MOVE_FORWARD when MOVE_LEFT/MOVE_RIGHT is not possible in order not to accelerate to avoid case distinction in step function?

Reply:
Missed the stopped aspect, well spotted!

My goal was to avoid the awkward test:
```elif preprocessed_action == RailEnvActions.MOVE_FORWARD and raw_action == RailEnvActions.MOVE_FORWARD:
                # accelerate, but not if left/right corrected to forward
                new_speed += self.acceleration_delta
```

Should we change design (to not "fixing" actions in the env) and fall back to DO_NOTHING. In case of symmetric switch, DO_NOTHING is undefined... So to STOPPED and penalize?